specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,247 @@
+# AppLifecycle - State Notifications
```

Comment:
Note on file paths - Maybe move this under `specs/AppLifecycle` directly so it's `specs/AppLifecycle/PowerStateNotification.md` or similar.